drivers/hmis/app/graphql/mutations/delete_client_file.rb

```diff
@@ -0,0 +1,23 @@
+module Mutations
+  class DeleteClientFile < BaseMutation
+    argument :file_id, ID, required: true
+
+    field :client, Types::HmisSchema::Client, null: true
+
+    def resolve(file_id:)
+      file = Hmis::File.find_by(id: file_id)
+
+      errors = HmisErrors::Errors.new
+      errors.add :file_id, :not_found unless file.present?
+      errors.add :file_id, :not_allowed if file.present? && !current_user.can_edit_clients_for?(file.client)
+      return { errors: errors } if errors.any?
+
+      file.destroy!
+
+      {
+        client: file.client,
+        errors: [],
+      }
+    end
```

Comment:
I think this can all be replaced with the common delete code. There shouldn't be anything special about deleting Files.
```suggestion
     default_delete_record(
        record: file,
        field_name: :file,
        permissions: : can_manage_client_files,
      )
   end
```